### PR TITLE
feat(docs-link): add env vars to link to downstream docs

### PR DESCRIPTION
**Pull request by craicoverflow**
https://issues.jboss.org/browse/AEROGEAR-9630

To verify:

1. `make cluster/prepare` to prepare the cluster.
2. In [operator.yaml](https://github.com/aerogear/mobile-developer-console-operator/blob/842a58a09d8d0cc950ab2dc4c02c11dcc39709d0/deploy/operator.yaml) modify the following values, setting them to some arbitrary URLs.

- UPS_DOCUMENTATION_URL
- IDM_DOCUMENTATION_URL
- SYNC_DOCUMENTATION_URL
- MSS_DOCUMENTATION_URL

3. `make install-operator` to install the operator.
4. `make install-mdc` to install the MDC application.
5. In OpenShift, look at the environment variables in the MDC application. They should refect those that were set in the operator.

Screenshot from 2019-07-23 12-59-50

There is no visual verification until https://github.com/aerogear/mobile-developer-console/pull/339 is merged.

### CODE REVIEW

File: deploy/operator.yaml
Comment (Boomatang) [inline deploy/operator.yaml]: Should `SYNC_DOCUMENTATION_URL` and `MSS_DOCUMENTATION_URL` not have double quotes around them like the other vars

**Comment by grdryn** [comment]
@craicoverflow is there a corresponding change to the integreatly installer to set the values for these env vars?

@wei-lee you previously spoke about a config file that was available to MDC where it could get this info. How does that relate to this?

File: deploy/operator.yaml
Comment (craicoverflow) [reply deploy/operator.yaml]: Good spot. They don't have to but it's best to keep consistent.

**Comment by craicoverflow** [comment]
> @craicoverflow is there a corresponding change to the integreatly installer to set the values for these env vars?

To be honest, I don't know the answer to that. I'm AFK today and @wei-lee is also, so may not be able to get an answer until Thursday.

> @wei-lee you previously spoke about a config file that was available to MDC where it could get this info. How does that relate to this?

I'm not sure if it does. Not yet anyway. The following snippet is an example of what the config file looks like.

```
[
  {
    "name": "Identity Management",
    "version": "7.3.2.GA",
    "host": "https://29493.openshiftworkshop.com",
    "type": "keycloak",
    "mobile": "true"
  },
  ...
]
```

**Comment by grdryn** [comment]
> > @wei-lee you previously spoke about a config file that was available to MDC where it could get this info. How does that relate to this?
> 
> I'm not sure if it does. Not yet anyway. The following snippet is an example of what the config file looks like.
> 
> ```
> [
>   {
>     "name": "Identity Management",
>     "version": "7.3.2.GA",
>     "host": "https://29493.openshiftworkshop.com",
>     "type": "keycloak",
>     "mobile": "true"
>   },
>   ...
> ]
> ```

Ah right, yes, this is the documentation URL, which is different from the service host. Ignore that question then :)

> > @craicoverflow is there a corresponding change to the integreatly installer to set the values for these env vars?
> 
> To be honest, I don't know the answer to that. I'm AFK today and @wei-lee is also, so may not be able to get an answer until Thursday.
> 

We can merge this and figure that out after :+1:

File: pkg/controller/mobiledeveloperconsole/mobiledeveloperconsole.go
Comment (aliok) [inline pkg/controller/mobiledeveloperconsole/mobiledeveloperconsole.go]: @craicoverflow can you read the env vars similar to here: https://github.com/aerogear/mobile-developer-console-operator/blob/842a58a09d8d0cc950ab2dc4c02c11dcc39709d0/pkg/config/config.go#L27

It is better to keep these env vars in a single place.

**Review by aliok — CHANGES_REQUESTED** [review]
@craicoverflow Just left some comments which should be easy to address.

Also, can you please add some documentation in the README? Similar to env vars listed here: https://github.com/aerogear/mobile-developer-console-operator/blob/842a58a09d8d0cc950ab2dc4c02c11dcc39709d0/README.adoc#configuration

File: pkg/controller/mobiledeveloperconsole/mobiledeveloperconsole.go
Comment (craicoverflow) [reply pkg/controller/mobiledeveloperconsole/mobiledeveloperconsole.go]: Sure yes, makes sense. I'm not sure why I didn't do this in the first place.

**Comment by laurafitzgerald** [comment]
@grdryn yes, you are right. There are changes required for the installer. Those vars can be added to the file here: https://github.com/integr8ly/installation/blob/master/roles/mdc/templates/operator.yaml.j2 Do we have the docs links @craicoverflow?

**Comment by craicoverflow** [comment]
> @grdryn yes, you are right. There are changes required for the installer. Those vars can be added to the file here: [integr8ly/installation:roles/mdc/templates/operator.yaml.j2@`master`](https://github.com/integr8ly/installation/blob/master/roles/mdc/templates/operator.yaml.j2) [](/integr8ly/installation/blob/HEAD@{2019-07-25T09:10:38Z}/roles/mdc/templates/operator.yaml.j2) Do we have the docs links @craicoverflow?

@laurafitzgerald yes I've sent you the links now.

File: pkg/config/config.go
Comment (grdryn) [inline pkg/config/config.go]: The second parameter to `getEnv` is for specifying a default that can be used if the env var isn't set. Is there a default that could be specified here?

Comment (laurafitzgerald) [reply pkg/config/config.go]: My thought is that we should default to the aerogear docs. This would be the default in the case of the community using the operator. We can override those in other cases. cc @finp

Comment (pwright) [reply pkg/config/config.go]: I don't understand the context here.

Comment (craicoverflow) [reply pkg/config/config.go]: @grdryn I left these blank because the default for each service is already set in the code. 

Example: https://github.com/aerogear/mobile-developer-console/blob/ca2f317336e891457c546e85ed30a9568460d2fe/src/models/mobileservices/pushvariantcr.js#L282

We could also add them as defaults here.

Comment (grdryn) [reply pkg/config/config.go]: Ah right, if they're there in the mdc already, that's enough for me :+1:

**Review by grdryn — APPROVED** [review]

File: pkg/config/config.go
Comment (laurafitzgerald) [reply pkg/config/config.go]: @craicoverflow I can provide links in here for you. I need to add some for downstream in the installer so we can do the same for upstream.

Comment (craicoverflow) [reply pkg/config/config.go]: Am I to wait for the links before merging this?

Comment (laurafitzgerald) [reply pkg/config/config.go]: @craicoverflow it's up to you. The links are there now in the linked pr. For example it would be https://docs.aerogear.org/limited-availability/upstream/sync.html

Comment (craicoverflow) [reply pkg/config/config.go]: thanks @laurafitzgerald - I've added the default values here now.

**Pull request merged**
